src/camvidlog2/vid.py

```
@@ -213,6 +214,69 @@ def generate_frames_cv2_rtsp(
+        )
+        try:
+            frame_thread.start()
+            while frame_thread.is_alive():
+                with lock:
+                    np.copyto(array2, array1)
+                yield array2
+        finally:
+            terminator.set()
+            frame_thread.join(5)
+            if frame_thread.is_alive():
+                raise RuntimeError("Frame thread did not terminate")
```

Ruff (0.11.9)

Avoid specifying long messages outside the exception class

(TRY003)